packages/packagers/html/src/HTMLPackager.js

```diff
@@ -112,6 +114,51 @@ async function getAssetContent(
   return null;
 }
 
+async function addIntrinsicImageSizes(
+  bundle: NamedBundle,
+  bundleGraph: BundleGraph<NamedBundle>,
+  tree,
+) {
+  const imageNodes = [];
+  tree.walk(node => {
+    if (node.tag === 'img' && node.attrs && node.attrs['src']) {
+      imageNodes.push(node);
+    }
+    return node;
+  });
+
+  for (const node of imageNodes) {
+    const dependency = bundle
+      .getMainEntry()
+      ?.getDependencies()
+      .find(bundle => bundle.id === node.attrs['src']);
+
+    if (!dependency) {
+      continue;
+    }
+
+    const image = bundleGraph.getResolvedAsset(dependency);
+
+    if (!image) {
+      continue;
+    }
+
+    if (
+      typeof image.meta.width !== 'number' &&
+      typeof image.meta.height !== 'number'
+    ) {
+      const size = imageSize(await image.getBuffer());
+      image.meta.width = size.width;
+      image.meta.height = size.height;
+    }
+
+    node.attrs.width = image.meta.width;
+    node.attrs.height = image.meta.height;
```

Comment:
I think we should only do this if a `width` or `height` attribute is not already set on the element?

Reply:
> I think we should only do this if a `width` or `height` attribute is not already set on the element?

Probably. I left this in for the moment (https://github.com/parcel-bundler/parcel/pull/6783#discussion_r697663129) for feedback (primarily about the CSS issue), but in the longer term it would be better for these to be "fixed" earlier in the pipeline (refs https://github.com/parcel-bundler/parcel/issues/3738#issue-516907103).